lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -277,6 +281,26 @@ def _get_credentials(self):
 
         return boto_params
 
+    def _boto3_assume_role(self, credentials, region):
+        """
+        Assume an IAM role passed by iam_role_arn parameter
+
+        :return: a dict containing the credentials of the assumed role
+        """
+
+        iam_role_arn = self.iam_role_arn
+
+        try:
+            sts_connection = boto3.session.Session(profile_name=self.boto_profile).client('sts', region, **credentials)
+            sts_session = sts_connection.assume_role(RoleArn=iam_role_arn, RoleSessionName='ansible_aws_ec2_dynamic_inventory')
+            return dict(
+                aws_access_key_id=sts_session['Credentials']['AccessKeyId'],
+                aws_secret_access_key=sts_session['Credentials']['SecretAccessKey'],
+                aws_session_token=sts_session['Credentials']['SessionToken']
+            )
+        except botocore.exceptions.ClientError as e:
+            raise AnsibleError("Unable to assume IAM role: %s" % to_native(e))
+
```

Comment:
@s-hertel 

talking with @AlanCoding and thinking about use of the `boto_profile`, which requires there be a file laid down on the system that is running `ansible-inventory`, I think this is much better for our use case in awx. While i understand they have overlapping functionality, using this is much less hacky than requiring people lay down files on every awx node that may run inventory update. This way we can provide credentials using standard aws credential and provide custom role to use that those credentials give access to.